From 7dddcc60477d6400794f4f4ba4bf64329be4e81d Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Wed, 14 Oct 2015 17:38:56 +0530
Subject: [PATCH 15/28] FOR_UPSTREAM [VPG]: drm/i915: Start link training from
 max link rate

DP spec requires link training to be performed from maximum link rate
supported by the panel. This patch enforces this logic.

Change-Id: Id0a91bc4bb4b1cae4752c04a906efac05edd792f
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9749
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/439688
---
 drivers/gpu/drm/i915/intel_dp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 59ff3fe..50a676f 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1064,7 +1064,7 @@ intel_dp_compute_config(struct intel_encoder *encoder,
 						   bpp);
 
 		for (lane_count = min_lane_count; lane_count <= max_lane_count; lane_count <<= 1) {
-			for (clock = min_clock; clock <= max_clock; clock++) {
+			for (clock = max_clock; clock >= min_clock; clock--) {
 				link_clock = drm_dp_bw_code_to_link_rate(bws[clock]);
 				link_avail = intel_dp_max_data_rate(link_clock,
 								    lane_count);
-- 
1.9.1

